src/wattsi.pas

```diff
@@ -1561,23 +1561,24 @@ function Split(const Document: TDocument; var BigTOC: TElement; const Base: Ansi
 
    procedure SaveFragmentLinks(const Data: Rope);
    var
-      F: Text;
+      LinkFixupFile, FragmentLinksFile: Text;
+      LinkFixup: String;
    begin
-      Assign(F, Base + 'fragment-links.js');
-      Rewrite(F);
-      Write(F, 'var fragment_links = {');
-      Write(F, Data.AsString);
-      Write(F, '};');
-      Write(F, 'var fragid = window.location.hash.substr(1);');
-      Write(F, 'if ((!fragid) || !(fragid in fragment_links)) {'); // handle section-foo.html links from the old old multipage version, and broken foo.html from the new version
-      Write(F,   'var m = window.location.pathname.match(/\/(?:section-)?([\w\-]+)\.html/);');
-      Write(F,   'if (m) fragid = m[1];');
-      Write(F, '}');
-      Write(F, 'var page = fragment_links[fragid];');
-      Write(F, 'if (page) {');
-      Write(F,   'window.location.replace(page+''.html#''+fragid);');
-      Writeln(F, '}');
-      Close(F);
+      // link-fixup.js
+      LinkFixup := ReadTextFile('html/link-fixup.js').AsString;
```

Comment:
I think we can’t assume thereÆs an `html` subdir with a `link-fixup.js` file.

We could instead use `GetEnvironmentVariable` to make wattsi read the value of `HTML_SOURCE` (directory for the clone of the `whatwg/html` repo) and get the `link-fixup.js` file from there—but `HTML_SOURCE` is not a wattsi thing, it’s a `whatwg/html-build` thing, so I think we should also not assume `HTML_SOURCE` is set.

Given all that I think the safest most-portable/robust assumption to build into wattsi is that there’s a `link-fixup.js` in the same directory as whatever file you feed to wattsi as the first parameter from the command line. That will the true if you run wattsi directly in a `whatwg/html` clone (which is something I think we should continue to support), so I think we should make `whatwg/html-build/build.sh` also ensure it’s true (by copying the `link-fixup.js` file from the `HTML_SOURCE` dir to the `HTML_TEMP` dir).

If we agree about all that, then please update the patch in the PR to do something like this:

```diff
diff --git a/src/wattsi.pas b/src/wattsi.pas
index ee51a1e..72c6519 100644
--- a/src/wattsi.pas
+++ b/src/wattsi.pas
@@ -1557,7 +1557,7 @@ begin
    Close(F);
 end;
 
-function Split(const Document: TDocument; var BigTOC: TElement; const Base: AnsiString): Boolean; // True = success, False = failed
+function Split(const Document: TDocument; var BigTOC: TElement; const Base: AnsiString; var LinkFixupJS: AnsiString): Boolean; // True = success, False = failed
 
    procedure SaveFragmentLinks(const Data: Rope);
    var
@@ -1565,7 +1565,7 @@ function Split(const Document: TDocument; var BigTOC: TElement; const Base: Ansi
       LinkFixup: String;
    begin
       // link-fixup.js
-      LinkFixup := ReadTextFile('html/link-fixup.js').AsString;
+      LinkFixup := ReadTextFile(LinkFixupJS).AsString;
       LinkFixup := StringReplace(LinkFixup, '/* WATTSI_INSERTS_FRAGMENT_LINKS_HERE */', Data.AsString, []);
       Assign(LinkFixupFile, Base + '../link-fixup.js');
       Rewrite(LinkFixupFile);
@@ -2209,6 +2209,7 @@ var
    ParamOffset: Integer = 0;
    Source: TFileData;
    OutputDirectory: AnsiString;
+   LinkFixupJS: AnsiString;
    Parser: THTMLParser;
    BigTOC: TElement;
    Documents: array[TVariants] of TDocument;
@@ -2230,6 +2231,12 @@ begin
          exit;
       end;
    OutputDirectory := ParamStr(2 + ParamOffset);
+   LinkFixupJS := ExtractFilePath(ParamStr(1 + ParamOffset)) + 'link-fixup.js';
+   if (not FileExists(LinkFixupJS)) then
+   begin
+      Writeln(LinkFixupJS + ' file not found.');
+      exit;
+   end;
    if (not IsEmptyDirectory(OutputDirectory)) then
    begin
       // only act if, when we start, the output directory is empty, to make sure that the
@@ -2311,7 +2318,7 @@ begin
                   {$IFDEF TIMINGS} Writeln('Splitting spec...'); {$ENDIF}
                   {$IFDEF TIMINGS} StartTime := Now(); {$ENDIF}
                   MkDir(OutputDirectory + '/multipage-' + kSuffixes[Variant]);
-                  if (not Split(Documents[Variant], BigTOC, OutputDirectory + '/multipage-' + kSuffixes[Variant] + '/')) then
+                  if (not Split(Documents[Variant], BigTOC, OutputDirectory + '/multipage-' + kSuffixes[Variant] + '/', LinkFixupJS)) then
                      raise EAbort.Create('Could not split specification');
                   {$IFDEF TIMINGS} Writeln('Elapsed time: ', MillisecondsBetween(StartTime, Now()), 'ms'); {$ENDIF}
                end;
```